### PR TITLE
Feature/contribution guide

### PR DESCRIPTION
**Pull request by CODER591**
---
name: Feature proposal or request

---
## Headline
Creation of Contribution Guide
## Summary
Creation of a Contribution guide to help external contributors get easier into the process, and also help existing contributors to find out useful information.
## Pro et contra
Creation of a Contribution guide will help external contributors get easier into the process, and also help existing contributors to find out useful information.
## Describe alternatives you've considered
No alternatives
## Additional context
Nothing here.

### CODE REVIEW

**Comment by edburmai** [comment]
Please squash the commits.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+* [Suggesting Enhancements](#suggesting-enhancements)
+* [Documentation Maintenance](#documentation-maintenance)
+
+
+### Reporting Vulnerabilities
+
+If You discovered the vulnerability:
+  1. DO NOT create an issue on GitHub, even with the pull request.
+     Help us avoid publicly known security issues.
+  2. Please, send us a detailed vulnerability report on the following e-mail:
```
Comment (edburmai) [inline CONTRIBUTING.md:44]: Seems something's missed.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+* **List some other applications where this enhancement exists.**
+* **Specify which version of Gateway You're using.** You can get a version from special config file
+* **Specify the name and version of the OS You are using.**
+
+### Pull Requests
+
+Here the checklist of what You need to create a successful pull request:
+1. Point issue on the repository
+2. Descriptive commit messages. [Checkout the guide](#git-commit-messages)
+3. In case You have changed the behaviour of some feature/ widely used function, etc....
```
Comment (edburmai) [inline CONTRIBUTING.md:114]: Contributors must change a documentation file located in each directory if a feature introduces considerable functionality, new concepts or design changes. We need to define the approach to follow regarding the documentation file in each code directory.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+     1. Provide a justification for this change
+     2. Change according documentation that listed in README in the folder with code.
+4. There is no need to assign reviewers to Your PR, we will do it for You.
+5. Pay attention to reviewers suggestions.
+6. Pull request will merged if and only if there is 3 approvals on it
+
+### Documentation Maintenance
+
+Firstly, [checkout out the documentation guide](#documentation-styleguide).
```
Comment (edburmai) [inline CONTRIBUTING.md:123]: I'm not really getting what is meant under this.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+* [Documentation Maintenance](#documentation-maintenance)
+
+
+### Reporting Vulnerabilities
+
+If You discovered the vulnerability:
+  1. DO NOT create an issue on GitHub, even with the pull request.
+     Help us avoid publicly known security issues.
+  2. Please, send us a detailed vulnerability report on the following e-mail:
+But, if You have a fix that You want to propose, send us an e-mail with:
```
Comment (edburmai) [inline CONTRIBUTING.md:45]: The same as above

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+  * [Documentation Styleguide](#documentation-styleguide)
+
+[Contribution Guide Thanks](#contribution-guide-thanks)
+
+## Code of Conduct
+
+1. Be nice and friendly
+2. Avoid any kind of abuses or strong language.
+3. Everything that You want to deliver to our community should be in well structured and polite form.
+4. Always make clear build before Pull Request
```
Comment (edburmai) [inline CONTRIBUTING.md:26]: IMHO, PR stuff is not something need to be defined in **Code of Conduct**

File: TEMPLATES/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+---
+Type: Bug/ Vulnerability report
+Severity: How it is severe?
```
Comment (anantonyk) [inline TEMPLATES/bug_report.md:3]: how *is it* severe?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+* [Use this guide to write acceptable commit message](https://wiki.openstack.org/wiki/GitCommitMessages)
+
+### Documentation Styleguide
+
+
+### Additional notes
+
+
+#### Thanks
```
Comment (anantonyk) [inline CONTRIBUTING.md:141]: It`s better to say "gratitude" or "acknowledgment" :-)

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+### Documentation Maintenance
+
+Firstly, [checkout out the documentation guide](#documentation-styleguide).
+
+### Styleguides
+
+* [Google C++ styleguide](https://google.github.io/styleguide/cppguide.html)
+* [Documentation Styleguide](#documentation-styleguide)
+* [Git Commit Messages](#git-commit-messages)
```
Comment (anantonyk) [inline CONTRIBUTING.md:129]: Don`t need to add *git commit messages* here, you placed it in next section.

File: CONTRIBUTING.md (hunk 7 above)
Comment (CODER591) [reply CONTRIBUTING.md:141]: Not sure regarding "acknowledgment", I guess You have pointed this word to the "Additional notes" phrase?

File: CONTRIBUTING.md (hunk 2 above)
Comment (CODER591) [reply CONTRIBUTING.md:114]: I mentioned these 2 lines under, but I would make it more descriptive.
But, I think your comment would be more applicable to the "Documentation maintenance" chapter.

File: CONTRIBUTING.md (hunk 3 above)
Comment (CODER591) [reply CONTRIBUTING.md:123]: There should be information regarding the maintenance of documentation files.
I mean general rules.

File: CONTRIBUTING.md (hunk 8 above)
Comment (CODER591) [reply CONTRIBUTING.md:129]: Let it be, it is subtable of contents table.
I mean, even book writers are pointing the "Introduction" point in the table of contents.

File: CONTRIBUTING.md (hunk 2 above)
Comment (CODER591) [reply CONTRIBUTING.md:114]: In any case, the "Documentation Maintenance" chapter, is all about approach.
I believe we should apply one approach to all documentation files.

File: CONTRIBUTING.md (hunk 1 above)
Comment (CODER591) [reply CONTRIBUTING.md:44]: All e-mails will be added later, as soon as they will be created.

File: CONTRIBUTING.md (hunk 5 above)
Comment (CODER591) [reply CONTRIBUTING.md]: I agree. Fixed.

**Pull request closed without merging**